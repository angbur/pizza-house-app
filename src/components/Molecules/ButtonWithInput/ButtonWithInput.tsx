import Button from 'components/Atoms/Button/Button';
import Icon from 'components/Atoms/Icon/Icon';
import QuantityChangeButton from 'components/Atoms/QuantityChangeButton/QuantityChangeButton';
import { ChangeEventHandler, MouseEventHandler } from 'react';
import styled from 'styled-components';

const StyledInput = styled.input`
  font-family: 'Source Sans Pro';
  font-size: 1rem;
  font-weight: 400;
  width: 40px;
  height: 43px;
  text-align: center;
  &:focus-visible {
    outline: none;
  }
  border-radius: 0;
  border: none;
  border-left: 1px solid black;
  border-right: 1px solid black;
`;

const StyledDiv = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  & > button {
    border-radius: 8px 0 0 8px;
    & > img {
      margin-right: 10px;
    }
  }
  & > input[type='number']::-webkit-inner-spin-button,
  & > input[type='number']::-webkit-outer-spin-button {
    -webkit-appearance: none;
    -moz-appearance: none;
    -os-appearance: none;
  }
  & > input[type='number'] {
    -moz-appearance: textfield;
  }
`;

type ButtonWithInputProps = {
  value: number;
  onIncrement?: MouseEventHandler<HTMLButtonElement>;
  onDecrement?: MouseEventHandler<HTMLButtonElement>;
  onClick?: MouseEventHandler<HTMLButtonElement>;
  onChange?: ChangeEventHandler<HTMLInputElement>;
};

const ButtonWithInput = ({
  value,
  onIncrement,
  onDecrement,
  onClick,
  onChange,
}: ButtonWithInputProps) => {
  return (
    <StyledDiv>
      <Button variant='secondary-dark' size={'sm'} onClick={onClick}>
        {Icon({ name: 'basket', size: 17, isActive: false })}
        Add to order
      </Button>
      <StyledInput type='number' value={value} onChange={onChange} />
      <QuantityChangeButton onIncrement={onIncrement} onDecrement={onDecrement} />
    </StyledDiv>
  );
};

export default ButtonWithInput;
