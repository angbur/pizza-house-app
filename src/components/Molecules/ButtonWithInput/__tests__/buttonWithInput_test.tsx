import renderer from 'react-test-renderer';
import ThemeProvider from 'components/Theme/ThemeContext';
import ButtonWithInput from '../ButtonWithInput';

const value = 1;

const Wrapper = () => {
  return (
    <ThemeProvider>
      <ButtonWithInput value={value} />
    </ThemeProvider>
  );
};

describe('Button with input', () => {
  test('should renders button with input correctly', () => {
    const tree = renderer.create(<Wrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
