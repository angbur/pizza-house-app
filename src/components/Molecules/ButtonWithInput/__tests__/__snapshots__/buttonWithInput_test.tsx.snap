// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button with input should renders button with input correctly 1`] = `
<div
  className="sc-ipEyDJ fkvujW quantity-button"
>
  <button
    className="sc-bcXHqe sc-eDvSVe feNVH cAJNIx secondary-dark btn-sm button"
    style={
      Object {
        "fontSize": "0.8500000000000001rem",
        "padding": "0.52rem 1rem",
      }
    }
  >
    <img
      height={17}
      src={
        Object {
          "ReactComponent": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "__esModule": true,
          "default": "basket.svg",
        }
      }
    />
    Add to order
  </button>
  <input
    className="sc-kDvujY dYXGJW"
    type="number"
    value={1}
  />
  <div
    className="sc-jrcTuL ggyCiw quantity-button"
  >
    <button>
      <img
        height={14}
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "arrow-up-active.svg",
          }
        }
      />
    </button>
    <button>
      <img
        height={14}
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "arrow-down-active.svg",
          }
        }
      />
    </button>
  </div>
</div>
`;
